index.js

```diff
@@ -149,24 +149,45 @@ class ServerlessCustomDomain {
       let nameLength = 0;
       // The arn of the choosen certificate
       let certificateArn;
+      // The certificate name
+      let certificateName = this.serverless.service.custom.customDomain.certificateName;
 
-      data.CertificateSummaryList.forEach((certificate) => {
-        let certificateListName = certificate.DomainName;
 
-        // Looks for wild card and takes it out when checking
-        if (certificateListName[0] === '*') {
-          certificateListName = certificateListName.substr(1);
+      // Checks if a certificate name is given
+      if (certificateName != null) {
+        if (certificateName.trim() === '') {
+          throw Error('certificateName is empty. Please put a valid certificate name or remove certificateName from the serverless file.');
         }
 
-        // Looks to see if the name in the list is within the given domain
-        // Also checks if the name is more specific than previous ones
-        if (this.givenDomainName.includes(certificateListName)
-          && certificateListName.length > nameLength) {
-          nameLength = certificateListName.length;
-          certificateArn = certificate.CertificateArn;
+        const foundCertificate = data.CertificateSummaryList
+          .find(certificate => (certificate.DomainName === certificateName));
```

Comment:
I think the intent is that the user specify the domain name of the certificate, rather than the ARN. In which case, validating that a cert for that domain name exists is good.